### PR TITLE
Update documentation on bell_inequality.py example

### PR DESCRIPTION
**Pull request by SamarthVadia**

### CODE REVIEW

**Comment by SamarthVadia** [comment]
Hi @vtomole 
Check the documentation update on bell inequality example as discussed in #3032. Thanks

**Comment by vtomole** [comment]
Thanks for opening this @SamarthVadia , I'll review it tomorrow.

File: examples/bell_inequality.py [hunk 1]
```diff
@@ -1,25 +1,56 @@
...
+The goal is to maximize the probability to satisfy the condition [2]:
+    a XOR b = x AND y
+
+In the classical deterministic case, the highest probability
+achievable is 75%. While with quantum correlations, it can
+achieve higher success probability. In the quantum case, two players
+Alice and Bob starts with a shared Bell-pair entangled state. The
+random input x and y is provided by referee for Alice and Bob. The
+success probability of satisfying the above condition will be
+cos(theta/2)^2 if Alice and Bob measures their entangled qubit in
```
Comment (vtomole) [inline examples/bell_inequality.py:17]: measures -> measure

File: examples/bell_inequality.py [hunk 2]
```diff
@@ -1,25 +1,56 @@
-"""Creates and simulates a circuit equivalent to a Bell inequality test.
+"""
+Bell's theorem or inequality proves that entanglement based
+observations can't be reproduced with any local realist theory [1].
+
+This example shows Bell inequality in form of CHSH game where two
```
Comment (vtomole) [inline examples/bell_inequality.py:5]: Bell -> Bell's

File: examples/bell_inequality.py [hunk 3]
```diff
@@ -1,25 +1,56 @@
...
+players Alice and Bob receive an input bit x and y respectively and
+produce an output a and b based on the input bit.
+The goal is to maximize the probability to satisfy the condition [2]:
+    a XOR b = x AND y
+
+In the classical deterministic case, the highest probability
+achievable is 75%. While with quantum correlations, it can
+achieve higher success probability. In the quantum case, two players
+Alice and Bob starts with a shared Bell-pair entangled state. The
```
Comment (vtomole) [inline examples/bell_inequality.py:14]: starts -> start

File: examples/bell_inequality.py [hunk 4]
```diff
@@ -1,25 +1,56 @@
...
+random input x and y is provided by referee for Alice and Bob. The
+success probability of satisfying the above condition will be
+cos(theta/2)^2 if Alice and Bob measures their entangled qubit in
+measurement basis v and w where angle between v and w is theta.
```
Comment (vtomole) [inline examples/bell_inequality.py:18]: Capitalize "v" and "w".

File: examples/bell_inequality.py [hunk 5]
```diff
@@ -1,25 +1,56 @@
...
+Therefore, maximum success probability is cos(pi/8)^2 ~ 85.3%
+when theta = pi/4.
+
+In the usual implementation [2], Alice and Bob shares the Bell state
```
Comment (vtomole) [inline examples/bell_inequality.py:22]: shares -> share

File: examples/bell_inequality.py [hunk 6]
```diff
@@ -1,25 +1,56 @@
...
+with same value and opposite phase. If the input x (y) is 0, Alice (Bob)
```
Comment (vtomole) [inline examples/bell_inequality.py:23]: "with the"

**Comment by SamarthVadia** [comment]
@vtomole Thanks for the quick feedback, changes are done!

File: examples/bell_inequality.py [hunk 7]
```diff
@@ -102,4 +133,4 @@ def bitstring(bits):
 
 
 if __name__ == '__main__':
-    main()
+    main()
```
Comment (vtomole) [inline examples/bell_inequality.py:136]: Revert this change

**Review by vtomole — APPROVED** [review]
Looks good to me, just one minor change needed.

File: examples/bell_inequality.py (hunk 7 above)
Comment (SamarthVadia) [reply examples/bell_inequality.py]: Done
